mungegithub/mungers/release-note-label.go

```diff
@@ -144,20 +144,15 @@ func (r *ReleaseNoteLabel) Munge(obj *github.MungeObject) {
 		return
 	}
 
+	releaseNoteLabelNeededAdded := false
 	if !obj.HasLabel(releaseNoteLabelNeeded) {
 		obj.AddLabel(releaseNoteLabelNeeded)
+		releaseNoteLabelNeededAdded = true
 	}
 
-	if !obj.HasLabel(lgtmLabel) {
-		return
+	if releaseNoteLabelNeededAdded && obj.HasLabel(lgtmLabel) {
+		obj.WriteComment(releaseNoteBody)
 	}
-
-	if obj.HasLabel(doNotMergeLabel) {
-		return
-	}
-
-	obj.WriteComment(releaseNoteBody)
-	obj.AddLabel(doNotMergeLabel)
```

Comment:
Okay, I see what you mean. The trip wire to ensure that this function didn't call multiple times was if this function ran all the way to the end, the PR had `do-not-merge` added to it.

The issue I'm seeing with the current logic is the `releaseNoteLabelNeededAdded` variable won't carry over multiple runs of this function. So we'll still be in a place that if the function is called, it will add the note over and over.